services/datalinker/templates/ingress-anonymous.yaml

```diff
@@ -0,0 +1,23 @@
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: {{ include "datalinker.fullname" . }}-anonymous
+  labels:
+    {{- include "datalinker.labels" . | nindent 4 }}
+  annotations:
+    {{- with .Values.ingress.annotations }}
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
+spec:
+  ingressClassName: "nginx"
+  rules:
+    - host: {{ required "global.host must be set" .Values.global.host | quote }}
+      http:
+        paths:
+          - path: "/api/hips/list"
```

Comment:
This is a completely random comment that isn't really about this, but is only tangentially related.  If we mess up the ingress and go to say /api/hipz, then we don't get routed through this ingress rule (obviously) but also we end up getting caught by the homepage I think.  This can make things confusing with apis and command line clients when they get a good helping of javascript with a 200.  But I'm not sure how to fix that, and this is all fine.

Reply:
It'd also be good if we could say for all of /api if you aren't authenticated to give back a 403 rather than an auth redirect too - we always forget that one, and that also is another common confuzzlement I have

Reply:
Yeah, agreed that we need somewhat better handling of 404s. I think that's something squareone should handle as part of being the default service. For the record, though, it does return a 404, not a 200, confirmed with `curl` just now (but yes, the page is full of JavaScript).

I had intentionally made `/api/hips` support the login redirect because unlike most API services I thought people were going to look at it in a web browser, but so far we've chosen not to publish an `index.html` and are using the Portal as the HiPS viewer instead of an embedded Aladin Lite instance. So I'll remove the redirect configuration for now.